### ISSUE TITLE
Port `twisted.internet.tcp` to Python 3

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6002]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)| @itamarst reported|
|-|-|
|Trac ID|trac#6002|
|Type|enhancement|
|Created|2012-09-18 15:23:28Z|
|Branch|https://github.com/twisted/twisted/tree/tcp-py3-6002-3|

`twisted.internet.tcp` should work on Python 3.

When doing this, tcp imports in `twisted.internet.posixbase` should be moved back up to the top of the module.

<details><summary>Searchable metadata</summary>

```
trac-id__6002 6002
type__enhancement enhancement
reporter__itamar itamar
priority__normal normal
milestone__Python_3_3_Minimal Python 3.3 Minimal
branch__branches_tcp_py3_6002_3 branches/tcp-py3-6002-3
branch_author__itamarst itamarst
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1347981808000000 1347981808000000
changetime__1350314333000000 1350314333000000
version__None None
owner__itamar itamar

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_1"></a>|@itamarst commented|
|-|-|

(In [36034]) Branching to 'tcp-py3-6002'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_2"></a>|@itamarst set owner to @itamarst|
|-|-|

This only covers `twisted.test.test_tcp` tests; the rest of the tests will be in [#6069](https://github.com/twisted/twisted/issues/6069).

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_3"></a>|@itamarst commented|
|-|-|

(In [36131]) Branching to 'tcp-py3-6002-3'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_4"></a>|@itamarst commented|
|-|-|

(In [36131]) Branching to 'tcp-py3-6002-3'

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_5"></a>|@itamarst set owner to @exarkun|
|-|-|

Ready for review, I think. Started http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/tcp-py3-6002-3

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_6"></a>|@exarkun set owner to @itamarst|
|-|-|

I noticed that `twisted.test.test_tcp.LargeBufferTestCase` is massively slower (~40x) on Python 3.  I guess we'll address that later, though (file a ticket if you feel like it, or we can just wait for someone to complain about it).  So: looks good, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="itamarst's avatar" src="https://avatars.githubusercontent.com/u/3266662?s=50" width="50" height="50">](https://github.com/itamarst)<a name="note_7"></a>|@itamarst set status to `closed`|
|-|-|

(In [36154]) Merge tcp-py3-6002-3.

Author: itamar
Review: exarkun
Fixes: [#6002](https://github.com/twisted/twisted/issues/6002)

Port twisted.internet.tcp to Python 3.
